SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java

```diff
@@ -659,7 +661,7 @@ private void cleanupRequestIfNoRemainingTasks(SingularityTaskCleanup cleanupTask
       requestManager.createCleanupRequest(
           new SingularityRequestCleanup(
               cleanupTask.getUser(), RequestCleanupType.DELETING, System.currentTimeMillis(),
-              Optional.of(Boolean.TRUE), requestId, Optional.<String> absent(),
+              Optional.of(Boolean.TRUE), Optional.absent(), requestId, Optional.<String> absent(),
```

Comment:
yeah, it would always be absent (and then default to true it looks like) for every request. You'd prob have to add the request option configuration to the the SingularityTaskCleanup since that's the only living memory of the request so you can do something like `cleanupTask.getRemoveFromLb`